doc/STYLE_GUIDE.md

```diff
@@ -0,0 +1,11 @@
+# MoveIt2 style guide
+
```

Comment:
Add note that the [MoveIt 1 style guide](https://moveit.ros.org/documentation/contributing/code/) is in effect, except where noted here in this md file